simple_network_sim/network_of_populations/__init__.py

```diff
@@ -854,13 +854,20 @@ def createNetworkOfPopulation(
     assert agesInInfectionMatrix == set(progression.keys()), "infection matrix and progression ages mismatch"
     assert agesInInfectionMatrix == {age for region in population.values() for age in region}, \
         "infection matrix and population ages mismatch"
-    assert set(graph.nodes()) == set(population.keys()), "regions mismatch between graph and population"
+    disconnected_nodes = set(population.keys()) - set(graph.nodes())
+    if disconnected_nodes:
+        logger.warning("These nodes are disconnected from the graph: %s", disconnected_nodes)
```

Comment:
If I'm being very picky, the error message "These nodes are disconnected from the graph:"  is probably not quite what we want - it's entirely possible for the graph to include small sets of disconnected vertices independent of the population, but this check isn't for that.  Maybe something more like "These nodes do not have any contacts in the specified network"